test_runner/regress/test_tenant_size.py

```diff
@@ -492,24 +505,41 @@ def get_current_consistent_size(
 
             collected_responses.append(("DELETE", current_lsn, size))
 
+        size_before_drop = get_current_consistent_size(
+            env, endpoint, size_debug_file, http_client, tenant_id, timeline_id
+        )[1]
+
         with endpoint.cursor() as cur:
             cur.execute("DROP TABLE t0")
 
-        # The size of the tenant should still be as large as before we dropped
-        # the table, because the drop operation can still be undone in the PITR
-        # defined by gc_horizon.
+        # Without setting a PITR interval, dropping the table doesn't reclaim any space
+        # from the user's point of view, because the DROP transaction is too small
+        # to fall out of gc_horizon.
         (current_lsn, size) = get_current_consistent_size(
             env, endpoint, size_debug_file, http_client, tenant_id, timeline_id
         )
-
         prev_size = collected_responses[-1][2]
-
         check_size_change(current_lsn, initdb_lsn, gc_horizon, size, prev_size)
 
+        # Set a tiny PITR interval to allow the DROP to impact the synthetic size
+        # Because synthetic size calculation uses pitr interval when available,
+        # when our tenant is configured with a tiny pitr interval, dropping a table should
+        # cause synthetic size to go down immediately
+        tenant_config["pitr_interval"] = "1ms"
```

Comment:
Aah yes, but this has to be non-zero for this to work.

Reply:
If update_gc_info would also return `last_record_lsn` in the case where `pitr_period == Duration::ZERO`, then this would work as expected using pitr=0.

Reply:
Discussed, this would had changed the value for existing configurations.